Cargo.toml

```diff
@@ -220,33 +221,46 @@ sha2                             = { version = "0.10" }
 sqlx                             = { version = "0.8", features = ["bigdecimal", "postgres", "runtime-tokio", "tls-rustls"] }
 tap                              = "1.0.1"
 tempfile                         = { version = "3.3.0" }
-tendermint                       = { default-features = false, version = "0.34.0" }
-tendermint-config                = { version = "0.34.0" }
-tendermint-light-client-verifier = { version = "0.34.0" }
-tendermint-proto                 = { version = "0.34.0" }
-tendermint-rpc                   = { version = "0.34.0" }
+tendermint                       = { default-features = false, version = "0.40.0" }
+tendermint-config                = { version = "0.40.0" }
+tendermint-light-client-verifier = { version = "0.40.0" }
+tendermint-proto                 = { version = "0.40.0" }
+tendermint-rpc                   = { version = "0.40.0" }
 termion                          = { version = "3" }
 thiserror                        = { version = "1.0" }
 time                             = { version = "0.3" }
-tokio                            = { version = "1.3" }
+tokio                            = { version = "1.39.0" }
 tokio-stream                     = { version = "0.1.8" }
 tokio-util                       = { version = "0.7" }
 toml                             = { version = "0.7" }
-tonic                            = { version = "0.10" }
-tonic-reflection                 = { version = "0.10.0" }
-tonic-web                        = { version = "0.10.0" }
+tonic                            = { version = "0.12.3" }
+tonic-reflection                 = { version = "0.12.3" }
+tonic-web                        = { version = "0.12.3" }
 tower                            = { version = "0.4.0" }
-tower-http                       = { version = "0.4" }
+tower-http                       = { version = "0.6.2" }
 tower-service                    = { version = "0.3.2" }
 tracing                          = { version = "0.1" }
 tracing-subscriber               = { version = "0.3.17", features = ["env-filter"] }
 url                              = { version = "2.2" }
 getrandom                        = { version = "0.2", default-features = false }
 
-# TODO(kate):
-# temporarily point these dependencies to a tag in the penumbra-zone fork.
-# see #4392, #4400, and hyperium/tonic#1701 for more information.
+# TODO: Don't use main; very bad bad. Use a rev.
+# TODO: wait for informalsystems to release a patch; see https://github.com/informalsystems/tendermint-rs/pull/1479.
 [patch.crates-io]
-tonic                            = { git = "https://github.com/penumbra-zone/tonic.git", tag = "v0.10.3-penumbra" }
-tonic-reflection                 = { git = "https://github.com/penumbra-zone/tonic.git", tag = "v0.10.3-penumbra" }
-tonic-web                        = { git = "https://github.com/penumbra-zone/tonic.git", tag = "v0.10.3-penumbra" }
+tendermint = { git = "https://github.com/informalsystems/tendermint-rs/", rev = "refs/pull/1480/head" }
```

Comment:
Merged, I shall point them to their main repo. We should create an issue to followup with their patch release.